zyte_common_items/base.py

```diff
@@ -38,30 +40,66 @@ class _ItemBase:
     __slots__ = ("_unknown_fields_dict",)
 
 
+def _get_import_path(obj: type):
+    return f"{obj.__module__}.{obj.__qualname__}"
+
+
+def _extend_breadcrumbs(breadcrumbs: Breadcrumbs, key: Union[int, str]):
+    if isinstance(key, str):
+        if not breadcrumbs:
+            breadcrumbs = key
+        else:
+            breadcrumbs += f".{key}"
+    else:
+        assert isinstance(key, int)
+        item = f"[{key}]"
+        if not breadcrumbs:
+            breadcrumbs = item
+        else:
+            breadcrumbs += item
+    return breadcrumbs
+
+
 @attrs.define
 class Item(_ItemBase):
     def __attrs_post_init__(self):
         self._unknown_fields_dict = {}
 
     @classmethod
-    def from_dict(cls, item: Optional[Dict]):
+    def from_dict(cls, item: Optional[Dict], *, breadcrumbs: Breadcrumbs = None):
```

Comment:
Is "breadcrumbs" argument something which end users should use? If not, it'd be probably better to make it _private.

Alternatively, we can have public from_dict which doesn't expose this argument, and then _from_dict method which has it. It might be a cleaner API, if breadcrumbs argument is only used internally, to generate error messages.